[DOCS] Collapse contents in cluster reroute docs (#54851)
elastic/elasticsearch: docs/reference/cluster/reroute.asciidoc

`POST /_cluster/reroute`

> [role="child_attributes"]

Reviewer: This attribute should be included in the sections containing the collapsible sections.
```suggestion
```

> ==== {api-query-parms-title}

Reviewer: ```suggestion
[role="child_attributes"]
==== {api-query-parms-title}
```

`dry_run`::
    (Optional, boolean) If `true`, then the request simulates the operation only
    and returns the resulting state.

`explain`::
    (Optional, boolean) If `true`, then the response contains an explanation of
    why the commands can or cannot be executed.

`metric`::
    (Optional, string) Limits the information returned to the specified metrics.
    Defaults to all but metadata The following options are available:

+

> .Properties of `metric`
> [%collapsible%open]
> ======

Reviewer: ```suggestion
Because these are query parameters, they're not objects _per se_. It's more accurate to describe them as `options.

I'm not sure we need the collapsible section here as these won't be nested more than one level deep.

.Options for `metric`
[%collapsible%open]
======
```

`routing_table`::
    Shows the `routing_table` part of the response.

`version`::
    Shows the cluster state version.
======

> =====
> ====

Reviewer: These block delimiters are unneeded and preventing the docs build from working properly.
```suggestion
```

>     (Required, object) Defines the commands to perform. Supported commands are:

Reviewer: GitHub won't allow me to add a comment, but `[role="child_attributes"]` needs to added directly above `==== {api-request-body-title}` or `[[cluster-reroute-api-request-body]]` around line 110.

>     (Required, object) Defines the commands to perform. Supported commands are:

Reviewer: ```suggestion
    (Required, array of objects) Defines the commands to perform. Supported commands are:
```

    Allocate an unassigned replica shard to a node. Accepts `index` and `shard`
    for index name and shard number, and `node` to allocate the shard to. Takes
    <<modules-cluster,allocation deciders>> into account.

> ======

Reviewer: I would move this delimiter _after_ the definition for `allocate_empty_primary`. `allocate_stale_primary` and `allocate_empty_primary` are properties for a `commands` object.

> =====
> ====

Reviewer: ```suggestion
```
